### PR TITLE
Added `squashWhitespace` utility test

### LINKED ISSUE #352: Add tests for `squashWhitespace` utility
Add a file `workspaces/util/src/__tests__/squashWhitespace-test.ts`, for testing the `squashWhitespace` utility.

To see some PRs that add tests, take a look at #288 and #290 for inspiration.

### PR DESCRIPTION
**Pull request by Ma-hawaj**
Closes #352.

### CODE REVIEW

**Review by elimanzo — COMMENTED** [review]
Hey @Ma-hawaj! Thanks for submitting a PR for these test. 

One suggestion I have is, It'd be really cool to see a test with new lines `\n` and tabs `\t`!

File: workspaces/util/src/__tests__/squashWhitespace-test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  it("removes whitespaces at the end of the string", () => {
+    expect(squashWhitespace("hello world        ")).toBe("hello world");
+  });
+
+  it("removes whitespaces at the beginginng of the string", () => {
+    expect(squashWhitespace("         hello world")).toBe("hello world");
+  });
+
+  it("nothing should be changed", () => {
```
Comment (miorel) [inline workspaces/util/src/__tests__/squashWhitespace-test.ts:18]: note: the string titles are sometimes used in test output, it's therefore helpful to have a sense of what is being tested from the title alone (without seeing the code)

so for example "removes whitespace at the end of the string" is very clear but "nothing should be changed" is not so clear without more context

File: workspaces/util/src/__tests__/squashWhitespace-test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    expect(squashWhitespace("hello world !")).toBe("hello world !");
+  });
+
+  it("empty string", () => {
```
Comment (miorel) [inline workspaces/util/src/__tests__/squashWhitespace-test.ts:22]: nit: all the other test titles start with a verb, let's update this one to do the same for consistency

File: workspaces/util/src/__tests__/squashWhitespace-test.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    expect(squashWhitespace("hello   world")).toBe("hello world");
+  });
+
+  it("removes whitespaces at the end of the string", () => {
+    expect(squashWhitespace("hello world        ")).toBe("hello world");
+  });
+
+  it("removes whitespaces at the beginginng of the string", () => {
+    expect(squashWhitespace("         hello world")).toBe("hello world");
+  });
```
Comment (miorel) [inline workspaces/util/src/__tests__/squashWhitespace-test.ts:16]: To echo @elimanzo it would be nice to have some tests that use other kinds of whitespace besides space. It would be nice to also test newlines (\n), tabs (\t), and even some of the more rare Unicode characters that are considered whitespace: https://en.wikipedia.org/wiki/Whitespace_character

File: workspaces/util/src/__tests__/squashWhitespace-test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { squashWhitespace } from "../squashWhitespace";
+
+describe("squashWhitespace", () => {
+  it("removes whitespaces between words if it exists", () => {
```
Comment (miorel) [inline workspaces/util/src/__tests__/squashWhitespace-test.ts:6]: nit: from what I've seen, "whitespace" is a collective noun, like "bread", so we don't have to add S to make it plural

please update here and below
```suggestion
  it("removes whitespace between words if it exists", () => {
```

File: workspaces/util/src/__tests__/squashWhitespace-test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  it("respects case", () => {
+    expect(squashWhitespace(" Hello   World  ")).toBe("Hello World");
+  });
+
+  it("works with emojis", () => {
+    expect(squashWhitespace("🍎   apple")).toBe("🍎 apple");
+    expect(squashWhitespace("🍌banana")).toBe("🍌banana");
+    expect(squashWhitespace("grapes 🍇 🍇   ")).toBe("grapes 🍇 🍇");
+  });
+});
```
Comment (miorel) [inline workspaces/util/src/__tests__/squashWhitespace-test.ts:35]: A couple more test cases to consider:

* a string that is entirely whitespace
* a string that has a single whitespace character that's not a space, e.g. "hello\nworld" --> "hello world"
* a string that has whitespace at the beginning, end, and repeated whitespace in the middle (not sure this was explicitly covered so far)

**Review by miorel — COMMENTED** [review]
Thanks for the pull request @Ma-hawaj, thisE looks solid! I added just a few comments.

Also, to confirm, are you comfortable contributing this code under the MIT License?

**Comment by Ma-hawaj** [comment]
yes, sure.
thanks for the comments @miorel @elimanzo 
I'll work on them.

**Review by miorel — APPROVED** [review]
This looks good to me @Ma-hawaj, thank you! Feel free to get in touch if you're interested in taking on something more significant, my email address is available in the `git log`.
